src/release-specification.ts

```diff
@@ -317,6 +318,55 @@ export async function validateReleaseSpecification(
         }
       }
 
+      // Check to compel users to release packages with breaking changes alongside their dependents
+      if (
+        versionSpecifierOrDirective === 'major' ||
+        (isValidSemver(versionSpecifierOrDirective) &&
+          diff(pkg.validatedManifest.version, versionSpecifierOrDirective) ===
+            'major')
+      ) {
+        const dependentNames = Object.keys(project.workspacePackages).filter(
+          (possibleDependentName) => {
+            const possibleDependent =
+              project.workspacePackages[possibleDependentName];
+            const { dependencies, peerDependencies } =
+              possibleDependent.validatedManifest;
+            return (
+              hasProperty(dependencies, packageName) ||
+              hasProperty(peerDependencies, packageName)
+            );
+          },
+        );
+        const missingDependentNames = dependentNames.filter((dependentName) => {
```

Comment:
Given the following scenario:

- There are two packages, A and B, where B depends on A
- A is listed with a major bump, but B is not listed

I think something might be missing: we're never checking whether there are any changes in B to release. 

Now... there's no reason why B _shouldn't_ have changes, because the version it's using for the A dependency should be bumped along with A. (If A is bumped from 1.0.0 to 2.0.0, then B's dependency on A should also be bumped to `~> 2.0.0`). However, I don't think we are ever enforcing that. That means it's possible for B to not have changes.

Should we add a new list here, similar to other lists we've created?

``` suggestion
        );
        const changedDependentNames = dependentNames.filter((dependentName) => {
          return project.workspacePackages[possibleDependentName]
            .hasChangesSinceLatestRelease;
        });
        const missingDependentNames = changedDependentNames.filter((dependentName) => {
```

Additionally, do you think it would be worth it to add some kind of check in a future PR which ensures that if workspace package A is being bumped to a major version X, and it is a major version, then all dependents / peer dependents of A are also bumped to `~^X.0.0`?